templates/_macros/vf_tiered-list.jinja

```diff
@@ -3,26 +3,90 @@
 #   desktop vs. 50/50 split between title/description
 # is_list_full_width_on_tablet: whether list title/description each have their
 #   own row on tablet vs. 50/50 split between title/description
+# is_media_full_width: whether the media is full width
+# media_placement: where the image is placed relative to the description. Options are "after_description" or "before_description".
 # Slots
 # title: top-level title element
 # description: top-level description element
+# video: top-level video element. Mutually exclusive with image, and takes priority over the image.
+# image: top-level image element. Mutually exclusive with video. Is not rendered if the video is provided.
 # list_item_title_[1-25]: title element of each child list item
 # list_item_description_[1-25]: description element of each child list item
 # cta: CTA block element
 {% macro vf_tiered_list(
   is_description_full_width_on_desktop=true,
-  is_list_full_width_on_tablet=true) -%}
+  is_list_full_width_on_tablet=true,
+  is_media_full_width=false,
+  media_placement="after_description"
+  ) -%}
   {%- set title_content = caller('title') -%}
   {%- set description_content = caller('description') -%}
   {%- set has_description = description_content|trim|length > 0 -%}
+  {%- set image_content = caller('image') -%}
+  {%- set has_image = image_content|trim|length > 0 -%}
   {%- set cta_content = caller('cta') -%}
   {%- set has_cta = cta_content|trim|length > 0 -%}
+  {%- set video_content = caller('video') -%}
+  {%- set has_video = video_content|trim|length > 0 -%}
+  {%- set has_media = has_image or has_video -%}
+
+  {% if media_placement not in ["before_description", "after_description"] %}
+    {% set media_placement = "after_description" %}
+  {% endif %}
+
+  {% set image_aspect_ratio = "16-9" %}
+  {% if is_media_full_width and not is_video %}
+    {% set image_aspect_ratio = "cinematic" %}
+  {% endif %}
+
+  {% macro _wrapped_media() %}
+    {%- if has_media -%}
+      {%- if has_video -%}
+        <div class="p-section--shallow">
+          {#- u-embedded-media applies 16:9 aspect ratio via padding. -#}
+          <div class="u-embedded-media">
```

Comment:
I see the application in being consistent with how other patterns work with specifying width, but splitting the 'block' and 'element' classes like this feels a little odd. I wonder how often people will forget to add the 'element' class _and_ if this will have a visual impact. If it doesn't, it wont get corrected and will promote incorrect use of BEM.

Reply:
The same for the video content below.